characters_counter/logging_config.py

```diff
@@ -0,0 +1,14 @@
+import logging
+
+
+def setup_logging() -> logging.Logger:
```

Comment:
The setup is good, but I assume you will be doing
```
logger = setup_logging()
```
In all files. Yes we have only one usage right now but imagine 10 files. Calling setup_logging() is not efficient to create Handlers all the time, set levers formatters etc.
Good practices would be

1.  At the bottom of the file do
```
logger = ...
```
And then make sure only that logger variable is in the init file (for future imports convenience).
You don't need setup_logging all over again, you configure it once and go on with it.
2. A common practice for naming the functions that do setup is `_factory` -> api_factory(), config_factory() - functions that return you either created object or possibility to create object. In your case, function returns object of the logging and you basically procude once logger instance and re-use it. I would suggest naming it accordingly.
3. Also, assuming we will be using just logger for imports, we don't want the setup logging to be that easily accessible. It's hard to hide things in python but there is a small nitpick you can use https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-single-and-double-underscore-before-an-object-name . Add leading underscore so people will get warning if they try to import the factory function that is meant not not be called

Reply:
I followed your recommendations and applied them to my code. They are very useful! I would be glad if you could review my code again when you have time.